nixos/modules/system/boot/tmp.nix

```diff
@@ -30,7 +30,13 @@ with lib;
 
   config = {
 
-    systemd.additionalUpstreamSystemUnits = optional config.boot.tmpOnTmpfs "tmp.mount";
+    systemd.mounts = mkIf config.boot.tmpOnTmpfs [
+      {
+        what = "tmpfs";
+        where = "/tmp";
+        mountConfig.Options = [ "mode=1777" "strictatime" "rw" "nosuid" "nodev" "size=50%" "huge=within_size" ];
```

Comment:
Can the `huge=within_size` be in a separate commit at least? It feels like a performance optimization that should be done on top of a fix (as in, if it causes regressions, I want to be able to revert it separately).

I'd even argue and say this is something very special that users should probably set on their own (or why is it not a default in general?)

Reply:
I removed the option altogether since people who want to use this functionality can just use the module systems's merge function to add options.